src/xcoobee/api/EventSubscriptionsApi.js

```diff
@@ -151,75 +88,120 @@ const deleteEventSubscription = (apiUrlRoot, apiAccessToken, arrayOfEventNames,
- * @param {CampaignId} campaignId - The campaign cursor.
+ * @param {string} referenceId - id of related enyity (i.e. campaignId)
+ * @param {string} referenceType - type of related entity (i.e. campaign, funding_panel)
  *
  * @returns {Promise<Object>} - The result.
- * @property {EventSubscription[]} data - A page of event subscriptions for the
- *   given campaign cursor.
- * @property {Object} page_info - The page information.
- * @property {boolean} page_info.has_next_page - Flag indicating whether there is
- *   another page of data to may be fetched.
- * @property {string} page_info.end_cursor - The end cursor.
+ * @property {EventSubscription[]} data - A page of event subscriptions for the user
  *
  * @throws {XcooBeeError}
  */
-const listEventSubscriptions = (apiUrlRoot, apiAccessToken, campaignId) => {
-  ApiUtils.assertAppearsToBeACampaignId(campaignId);
-  const query = `
-    query listEventSubscriptions($campaignId: String!) {
-      event_subscriptions(campaign_cursor: $campaignId) {
-        data {
-          campaign_cursor
-          date_c
-          event_type
-          handler
-          owner_cursor
-        }
-        page_info {
-          end_cursor
-          has_next_page
-        }
+const listEventSubscriptions = (apiUrlRoot, apiAccessToken, referenceId, referenceType) => {
+  const query = `{
+    event_subscriptions {
+      data {
+        owner_cursor
+        reference_cursor
+        reference_type
+        topic
+        channel
+        handler
+        date_c
       }
     }
-  `;
+  }`;
+
   return ApiUtils.createClient(apiUrlRoot, apiAccessToken).request(query, {
-    campaignId,
+    referenceId,
+    referenceType,
   })
-    .then((response) => {
-      const { event_subscriptions } = response;
+    .then((response) => response.event_subscriptions)
+    .catch((err) => {
+      throw ApiUtils.transformError(err);
+    });
+};
 
-      // TODO: Remove the following once `event_subscriptions` returns a valid `page_info`
-      // property or no longer does cursor-based pagination.
-      event_subscriptions.page_info = event_subscriptions.page_info || { end_cursor: null, has_next_page: false };
+/**
+ * Fetches available subscriptions and channels for the user
+ *
+ * @async
+ * @param {string} apiUrlRoot - The root of the API URL.
+ * @param {ApiAccessToken} apiAccessToken - A valid API access token.
+ * @param {string} referenceId - id of related enyity (i.e. campaignId)
+ * @param {string} referenceType - type of related entity (i.e. campaign, funding_panel)
+ *
+ * @returns {Promise<Array>} - All available subscriptions to subscribe.
+ *
+ * @throws {XcooBeeError}
+ */
+const getAvailableSubscriptions = (apiUrlRoot, apiAccessToken, referenceId, referenceType) => {
+  const query = `query getAvailableSubscriptions($referenceType: EventReferenceType, $referenceId: String){
+    available_subscriptions (reference_type: $referenceType, reference_cursor: $referenceId) {
+        topic
+        channels

```

Comment:
```suggestion
      topic
      channels
```
Please fix indentation in other places to 2 spaces too. Thanks.